app/CreateDeal.tsx

```diff
@@ -60,6 +65,21 @@ export function CreateDeal(props: {
                 />
               </div>
             </FormGroup>
+            <FormGroup>
+              <FormLabel>Unit</FormLabel>
+              {units.map((unit:Unit) => {
```

Comment:
usually you don't have to type the argument of the callback, typescript is smart enough to know that `unit` is of type `Unit` since `units` is of type `Unit[]`

This is fine also though